testsuite/MDAnalysisTests/analysis/test_lineardensity.py

```diff
@@ -47,53 +50,142 @@ def test_invalid_grouping():
+
+
+@pytest.fixture(scope="module")
+def testing_Universe():
+    """Generate a universe for testing whether LinearDensity works with
+       updating atom groups. Also used for parallel analysis test."""
+    n_atoms = 3
+    u = mda.Universe.empty(n_atoms=n_atoms,
+                           n_residues=n_atoms,
+                           n_segments=n_atoms,
+                           atom_resindex=np.arange(n_atoms),
+                           residue_segindex=np.arange(n_atoms))
+
+    for attr in ["charges", "masses"]:
+        u.add_TopologyAttr(attr, values=np.ones(n_atoms))
+
+    coords = np.array([
+                      [[1., 1., 1.], [1., 2., 1.], [2., 1., 1.]],
+                      [[1., 1., 2.], [1., 2., 1.], [2., 1., 1.]],
+                      [[1., 1., 3.], [1., 2., 1.], [2., 1., 1.]],
+                      [[1., 1., 4.], [1., 2., 1.], [2., 1., 1.]],
+                      [[1., 1., 5.], [1., 2., 1.], [2., 1., 1.]]
+                      ])
+
+    u.trajectory = get_reader_for(coords)(coords, order='fac', n_atoms=n_atoms)
+
+    for ts in u.trajectory:
+        ts.dimensions = np.array([2, 2, 6, 90, 90, 90])
+
+    return u
+
+
+def test_updating_atomgroup(testing_Universe):
+    expected_z_pos = np.array([0., 0.91329641, 0.08302695, 0., 0., 0.])
+    u = testing_Universe
+    selection = u.select_atoms("prop z < 3", updating=True)
+    ld = LinearDensity(selection, binsize=1).run()
+    assert_allclose(ld.results.z.mass_density, expected_z_pos)
+    # Test whether histogram bins are saved correctly.
+    expected_bin_edges = np.arange(0, 7)
+    assert_allclose(ld.results.x.hist_bin_edges, expected_bin_edges)
+
+
+testdict = {"pos": "mass_density",
+            "pos_std": "mass_density_stddev",
+            "char": "charge_density",
+            "char_std": "charge_density_stddev"}
+
+
+# TODO: Remove in 3.0.0
+def test_old_name_deprecations():
+    universe = mda.Universe(waterPSF, waterDCD)
+    sel_string = 'all'
+    selection = universe.select_atoms(sel_string)
+    ld = LinearDensity(selection, binsize=5).run()
+    with pytest.warns(DeprecationWarning):
+        assert_allclose(ld.results.x.pos, ld.results.x.mass_density)
+        assert_allclose(ld.results.x.pos_std, ld.results.x.mass_density_stddev)
+        assert_allclose(ld.results.x.char, ld.results.x.charge_density)
+        assert_allclose(ld.results.x.char_std,
+                        ld.results.x.charge_density_stddev)
+        for key in testdict.keys():
+            assert_allclose(ld.results["x"][key],
+                            ld.results["x"][testdict[key]])
+
+    # Check that no DeprecationWarning is raised with new attributes
+    with no_deprecated_call():
+        ld.results.x.mass_density
+        ld.results.x.mass_density_stddev
+        ld.results.x.charge_density
+        ld.results.x.charge_density_stddev
+
+
+# TODO: deprecated, remove in 3.0.0
+def test_parallel_analysis(testing_Universe):
```

Comment:
Can you add a `pytest.warns` somewhere here to check that the deprecation warning is being triggered (unless you're checking this somewhere else and I've missed it).